File: Leetcode/3304 - Find the K-th Character in String Game I/solution.ts
function kthCharacter(k: number): string {
    const findChar = (k: number): string => {
        if (k === 1) return 'a';

        let length = 1;
        while (length < k) {
            length *= 2;
        }

        const half = length / 2;

        if (k <= half) {
            return findChar(k);
        } else {
            const char = findChar(k - half);
            return String.fromCharCode(((char.charCodeAt(0) - 97 + 1) % 26) + 97);
        }
    };

    return findChar(k);
};

File: Leetcode/3304 - Find the K-th Character in String Game I/solution.py
def kthCharacter(self, k: int) -> str:
    def find_char(k: int) -> str:
        if k == 1:
            return 'a'

        length = 1
        while length < k:
            length *= 2

        half = length // 2

        if k <= half:
            return find_char(k)
        else:
            char = find_char(k - half)
            shifted = (ord(char) - ord('a') + 1) % 26
            return chr(ord('a') + shifted)

    return find_char(k)